src/main/java/wolox/training/models/Book.java

```diff
@@ -68,6 +70,8 @@ public String getImage() {
   }
 
   public void setImage(String image) {
+    Preconditions.checkNotNull(image);
```

Comment:
I would join these cases in the checkArgument, as follows:
```Preconditions.checkArgument(image != null && !image.isEmpty());```
Do that for every case where you've got checknotnull and checkargument.
What you did it's not wrong, of course.
But usually you use the checknotnull only if you're checking that fact and nothing else.